lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1253,10 +1268,16 @@ def create_default_nic(self):
                              "machine.".format(virtual_network_name)
 
             subnet_id = None
-            try:
-                subnets = self.network_client.subnets.list(self.resource_group, virtual_network_name)
-            except CloudError:
-                self.fail(no_subnets_msg)
+            if self.virtual_network_resource_group:
+                try:
+                    subnets = self.network_client.subnets.list(self.virtual_network_resource_group, virtual_network_name)
```

Comment:
I could have sworn I tried that and it wouldn't let me use a local variable but I didn't really understand how the module worked that well then. I'll test out the local variable and make the necessary changes.